src/test/java/org/jabref/logic/importer/fileformat/BibtexParserTest.java

```diff
@@ -1923,4 +1923,27 @@ void bibTeXConstantAprilIsParsedAsStringMonthAprilWhenReadingTheField() throws P
 
         assertEquals(Optional.of("#apr#"), result.get().getField(StandardField.MONTH));
     }
+
+    @Test
+    void conflictMarkersResultInProperParseError() throws Exception {
+        String input = """
+                @Article{first,
+                    author = {Author},
+                    <<<<<<< HEAD:test.bib
+                    title = {Title},
+                    =======
+                    title = {My title},
+                    >>>>>>> 77976da35a11db4580b80ae27e8d65caf5208086:test.bib
+                }
+
+                @Article{second,
+                  author = {Valid Author},
+                }
+                """;
+        ParserResult parserResult = parser.parse(new StringReader(input));
+
+        ParserResult expected = ParserResult.fromErrorMessage("Found git conflict markers");
+
+        assertEquals(expected, parserResult);
```

Comment:
I got the test passing by checking for git markers inside the `read` method after consuming a newline character.
I also called `checkForGitMarkers` at the beginning of the file to ensure it's called on the first line.

```java
private int read() throws IOException {
        int character = pushbackReader.read();

        if (!isEOFCharacter(character)) {
            pureTextFromFile.offerLast((char) character);
        }
        if (character == '\n') {
            line++;
            checkForGitConflictMarker();
        }
        return character;
    }
```

This is the logic. It looks for a line that starts with the 'ours' marker, which is represented by the symbol `<<<<<<<`. Then it continues to skip lines until it reaches the 'theirs' marker `>>>>>>>`. 

```java

    private void checkForGitConflictMarker() throws IOException {
        skipSpace();
        int markerCount = 0;
        // Looking for the 'ours' marker
        char c;
        while ((c = (char) peek()) == '<' && !isEOFCharacter(c)) {
            read();
            markerCount++;
        }
        if (markerCount == 7) {
            parserResult.addWarning("Found git conflict markers at line %d".formatted(line));
            // Skip 'ours' marker <<<<<<<
            skipLine();
            // Keep skipping lines until we hit the beginning of 'theirs' marker >>>>>>>
            while (peek() != '>' && !isEOFCharacter(peek())) {
                skipLine();
            }
            // Skip 'theirs' marker if we haven't hit EOF already
            if (!isEOFCharacter(peek())) {
                skipLine();
            }
        }
    }

    private void skipLine() throws IOException {
        while (peek() != '\n' && !isEOFCharacter(peek())) {
            read();
        }
        skipOneNewline();
    }
```
I had to modify the test slightly to pass because the logic I used would continue parsing after the marker, resulting in a parser result with two entries when the expected parser result is zero. I changed it to check whether the warning list contains the git conflict warning.
```java
       assertTrue(parserResult.warnings().contains("Found git conflict markers at line 3"));
```

Reply:
Where is the commit? 🎉👀

Reply:
I didn't make a commit. I got the inspiration of the solution while working on another PR, so I just made the changes on that PR's branch. However, you can use the code above inside BibTeXParser, and don't forget to set co-authored with me 😁.